python/ray/data/dataset.py

```diff
@@ -3394,6 +3394,13 @@ def write_bigquery(
                 overwritten if it exists.
             ray_remote_args: Kwargs passed to ray.remote in the write tasks.
         """
+        if ray_remote_args is None:
+            ray_remote_args = {}
+
+        # Each write task will launch individual remote tasks to write each block
+        # For correctness purposes, the write task cannot be retried
```

Comment:
Can you elaborate on "for correctness purposes"? Like, what incorrect behavior would you get if we did retries?

Reply:
There could be duplicate block writes. I added that to the comment.